### PR TITLE
[DB-12270] Categorize FOREIGN tables and report in analyze/assess about the required manual intervention

### PR DESCRIPTION
**Pull request by priyanshi-yb**
**Description:**

1. Categorize the FOREIGN TABLEs in a separate file called `foreign_table.sql` under `tables` but right now voyager doesn't dump the SERVER and USER MAPPING DDLs so reporting that in analyze and in asses as note.
Screenshot 2024-08-13 at 5 46 04 PM

3. Categorize ROW SECURITY type DDLs in TABLE 
example 
```
ALTER TABLE table_name ENABLE ROW LEVEL SECURITY;
```

Fixes - https://yugabyte.atlassian.net/browse/DB-12270
Tests - analyze tests are added but actual migration tests will be covered in pg sample schema's PR.

Jenkins - 
Oracle tests - https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/3130/

### CODE REVIEW

File: yb-voyager/cmd/analyzeSchema.go [hunk 1]
```diff
@@ -774,6 +795,9 @@ func getCreateObjRegex(objType string) (*regexp.Regexp, int) {
 	} else if objType == "TABLE" || objType == "PARTITION" {
 		createObjRegex = re("CREATE", opt("OR REPLACE"), "TABLE", ifNotExists, capture(ident))
 		objNameIndex = 3
+	} else if objType == "FOREIGN TABLE" {
+		createObjRegex = re("CREATE", opt("OR REPLACE"), "FOREIGN", "TABLE", ifNotExists, capture(ident))
```
Comment (makalaaneesh) [inline yb-voyager/cmd/analyzeSchema.go:799]: Slightly confused here. We are using both parser as well as regex?

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go [hunk 2]
```diff
@@ -135,12 +135,14 @@ func parseSchemaFile(exportDir string, schemaDir string, exportObjectTypesList [
 				objSqlStmts["SEQUENCE"].WriteString(stmts)
 			case "INDEX", "INDEX ATTACH":
 				objSqlStmts["INDEX"].WriteString(stmts)
-			case "TABLE", "DEFAULT", "CONSTRAINT", "FK CONSTRAINT", "CHECK CONSTRAINT":
+			case "TABLE", "DEFAULT", "CONSTRAINT", "FK CONSTRAINT", "CHECK CONSTRAINT", "ROW SECURITY":
```
Comment (makalaaneesh) [inline yb-voyager/src/srcdb/pg_dump_extract_schema.go:138]: What are we ultimately doing with ROW SECURITY? Do we want to categorize it as unsupported or something?

File: yb-voyager/cmd/analyzeSchema.go (hunk 1 above)
Comment (priyanshi-yb) [reply yb-voyager/cmd/analyzeSchema.go:799]: No no, this is for a different part of the code where we are using this regex to parse the stmt and get the basic info to prepare the `sqlInfo` object for further processing. I don't want to change this part of the code to parser yet for this case so just added regex.

Comment (priyanshi-yb) [reply yb-voyager/cmd/analyzeSchema.go:799]: But for reporting we are using Parser.

File: yb-voyager/src/srcdb/pg_dump_extract_schema.go (hunk 2 above)
Comment (priyanshi-yb) [reply yb-voyager/src/srcdb/pg_dump_extract_schema.go:138]: We are categorizing it as a part TABLE object as it is an ALTER TABLE but it is supported in YB so no need to report it as unsupported or anything.

Comment (makalaaneesh) [reply yb-voyager/src/srcdb/pg_dump_extract_schema.go:138]: Oh okay, so before it was going to uncategorized?

Comment (priyanshi-yb) [reply yb-voyager/src/srcdb/pg_dump_extract_schema.go:138]: right

File: migtests/tests/pg/assessment-report-test/expectedAssessmentReport.json [hunk 3]
```diff
@@ -317,6 +317,10 @@
 					"SqlStatement": "ALTER TABLE ONLY schema2.sales_region\n    ADD CONSTRAINT sales_region_pkey PRIMARY KEY (id, region);"
 				}
 			]
+		},
+		{
+			"FeatureName": "Foreign tables",
+			"Objects": []
```
Comment (makalaaneesh) [inline migtests/tests/pg/assessment-report-test/expectedAssessmentReport.json:323]: Why no objects?

File: yb-voyager/cmd/assessMigrationCommand.go [hunk 4]
```diff
@@ -809,6 +809,8 @@ func fetchUnsupportedPGFeaturesFromSchemaReport(schemaAnalysisReport utils.Schem
 	addUnsupportedFeaturesFromSchemaAnalysisReport("Conversion objects", CONVERSION_ISSUE_REASON, schemaAnalysisReport, &unsupportedFeatures)
 	addUnsupportedFeaturesFromSchemaAnalysisReport("Gin Indexes on Multi-columns", GIN_INDEX_MULTI_COLUMN_ISSUE_REASON, schemaAnalysisReport, &unsupportedFeatures)
 	addUnsupportedFeaturesFromSchemaAnalysisReport("Unsupported DDL operations", ADDING_PK_TO_PARTITIONED_TABLE_ISSUE_REASON, schemaAnalysisReport, &unsupportedFeatures)
+	addUnsupportedFeaturesFromSchemaAnalysisReport("Foreign tables", FOREIGN_TABLE_ISSUE_REASON, schemaAnalysisReport, &unsupportedFeatures)
```
Comment (makalaaneesh) [inline yb-voyager/cmd/assessMigrationCommand.go:812]: I don't think we should be classifying this as an unsupported feature. It is supported in YB. It's just that you need to configure the prerequisites yourself. 

Shall we just add a note to the assess-migration report saying something along the lines of - "you have some foreign tables. after running export-schema, you will have to do so and so manually. For more information, you can run analyze-schema."?

File: migtests/tests/pg/assessment-report-test/expectedAssessmentReport.json (hunk 3 above)
Comment (priyanshi-yb) [reply migtests/tests/pg/assessment-report-test/expectedAssessmentReport.json:323]: I haven't added a test in assessment test for that as it needs a table from another database and so on. So thought anyways @shubham-yb is adding sample schemas which has this so should be validated there

File: yb-voyager/cmd/assessMigrationCommand.go (hunk 4 above)
Comment (priyanshi-yb) [reply yb-voyager/cmd/assessMigrationCommand.go:812]: Hmm, that makes sense. Will make the change

Comment (priyanshi-yb) [reply yb-voyager/cmd/assessMigrationCommand.go]: Made the change and added sample in description

**Review by makalaaneesh — APPROVED** [review]
LGTM

**Pull request merged**